### ISSUE TITLE
It's possible to edit grains after logout when page is not refreshed

### ISSUE THREAD

**Issue by JamborJan** [sandstorm-io/sandstorm#186]
Scenario can be reproduced as follows:
- login to sandstorm
- open a grain (I tested it with ethercalc and wordpress)
- open a section which clearly needs a user e.g. edit a page in wordpress app
- click the logout button of sandstorm (not wordpress)
- you see that you are logged out from sandstorm but still logged in in wordpress
- you can edit and change stuff

Only a complete page refresh will make the grain realize that no user is logged in anymore.

Maybe it's a good idea to force the refresh after login / logout

JJ
![bildschirmfoto_2014-12-15_um_11_12_13](https://cloud.githubusercontent.com/assets/7508960/5433923/8cb7ef56-844c-11e4-801a-03c0b355f3e2.png)

**Comment by kentonv** [comment]
Yes, specifically: the session should be aborted, a new session should be started, and the iframe should be directed to that new session.

In fact, this should also happen if we detect that any of the session properties have changed, which include:
* `UserInfo` (from `grain.capnp`)
  * User identity.
  * Display name.
  * Permissions (can't change currently, but when it can, this will be the most important one to reset on!)
* `WebSessionParams` (from `web-session.capnp`)
  * `User-Agent` header.
  * `Accept-Languages` header.

**Comment by kentonv** [comment]
Keeping this open because we've only fixed the login/logout case, not the other ones I listed.
